src/Sulu/Bundle/AdminBundle/Controller/ContentNavigationController.php

```
@@ -0,0 +1,78 @@
+use FOS\RestBundle\Routing\ClassResourceInterface;
+use FOS\RestBundle\View\View;
+use FOS\RestBundle\View\ViewHandler;
+use FOS\RestBundle\View\ViewHandlerInterface;
+use Sulu\Bundle\AdminBundle\Navigation\ContentNavigationAliasNotFoundException;
+use Sulu\Bundle\AdminBundle\Navigation\ContentNavigationCollectorInterface;
+use Sulu\Component\Rest\Exception\RestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This controller returns all the content navigation items for a given alias
+ * @RouteResource("content-navigation")
+ */
+class ContentNavigationController implements ClassResourceInterface
+{
+    /**
+     * @var ContentNavigationCollectorInterface
+     */
+    private $contentNavigationCollector;
+
+    /**
+     * @var ViewHandler
+     */
+    private $viewHandler;
+
+    public function __construct(
+        ContentNavigationCollectorInterface $contentNavigationCollector,
+        ViewHandlerInterface $viewHandler
+    ) {
+        $this->contentNavigationCollector = $contentNavigationCollector;
+        $this->viewHandler = $viewHandler;
+    }
+
+    /**
+     * Returns all the content navigation items for a given alias
+     * @param Request $request
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        try {
+            $alias = $request->get('alias');
+
+            if (!$alias) {
+                throw new RestException('The alias attribute is required to load the content navigation');
+            }
+
+            $options = $request->query->all();
+
+            $contentNavigationItems = $this->contentNavigationCollector->getNavigationItems($alias, $options);
+
+            $view = View::create($contentNavigationItems);
+        } catch (ContentNavigationAliasNotFoundException $exc) {
+            $restException = new RestException(
+                sprintf('The content navigation alias "%s" does not exist!', $exc->getAlias())
```

But here you lose information. You hide the real exception.

The third argument to an exception is a `$previousException`. You should always pass the previous exception.

But I don't see any problem with a 500 here. This is a runtime error that should not happen.